letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -262,7 +276,15 @@ def choose_vhost(self, target_name):
             self.assoc[target_name] = vhost
             return vhost
 
-        return self._choose_vhost_from_list(target_name)
+        vhost = self._choose_vhost_from_list(target_name)
+        if vhost.ssl:
+            # remove duplicated or conflicting ssl directives
+            self._deduplicate_directives(vhost.path,
+                ["SSLCertificateFile", "SSLCertificateKeyFile"])
+            # remove all problematic directives
+            self._remove_directives(vhost.path, ["SSLCertificateChainFile"])
```

Comment:
In my testing, this `_remove_directives` call appears to not be working.  I created a test situation where my apache config contained these lines:
```
...
SSLCertificateFile /tmp/cert.pem
SSLCertificateKeyFile /tmp/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
SSLCertificateChainFile /tmp/chain.pem
...
```
Running the code from this branch (with master merged in), I got this resulting file:
```
<IfModule mod_ssl.c>
<VirtualHost *:443>
        # The ServerName directive sets the request scheme, hostname and port that
        # the server uses to identify itself. This is used when creating
        # redirection URLs. In the context of virtual hosts, the ServerName
        # specifies what hostname must appear in the request's Host: header to
        # match this virtual host. For the default virtual host (this file) this
        # value is not decisive as it is used as a last resort host regardless.
        # However, you must set it for any further virtual host explicitly.
        ServerName isnot.org

        ServerAdmin webmaster@localhost
        DocumentRoot /var/www/isnot

        # Available loglevels: trace8, ..., trace1, debug, info, notice, warn,
        # error, crit, alert, emerg.
        # It is also possible to configure the loglevel for particular
        # modules, e.g.
        #LogLevel info ssl:warn

        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined

        # For most configuration files from conf-available/, which are
        # enabled or disabled at a global level, it is possible to
        # include a line for only one particular virtual host. For example the
        # following line enables the CGI configuration for this host only
        # after it has been globally disabled with "a2disconf".
        #Include conf-available/serve-cgi-bin.conf
SSLCertificateFile /etc/letsencrypt/live/isnot.org-0001/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/isnot.org-0001/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
SSLCertificateChainFile /tmp/chain.pem
</VirtualHost>
# vim: syntax=apache ts=4 sw=4 sts=4 sr noet
</IfModule>
```
(And the resulting site is transvalid :( ).
```